aws_jupyter_proxy/awsproxy.py

```diff
@@ -43,6 +48,13 @@ def get_service_info(
         region_name=region,
     )
 
+    if (
+        endpoint_override
+        and validators.url(endpoint_override)
+        and re.compile(".*\.(aws.dev|amazonaws.com)$").match(endpoint_override)
```

Comment:
Is this going to cause issue with other aws partitions like CN?

Related nit: can we extract the regex `.compile` function into a constant?